File: app/src/main/java/com/example/rossinyamaral/bank/statementsScreen/StatementsRouter.java
package com.example.rossinyamaral.bank.statementsScreen;

import android.content.Intent;
import android.support.annotation.NonNull;
import android.view.View;
import android.widget.AdapterView;

import java.lang.ref.WeakReference;


interface StatementsRouterInput {
    public Intent navigateToSomeWhere(int position);

    public void passDataToNextScene(int position, Intent intent);
}

public class StatementsRouter implements StatementsRouterInput {

    public static String TAG = StatementsRouter.class.getSimpleName();
    public WeakReference<StatementsActivity> activity;


    @NonNull
    @Override
    public Intent navigateToSomeWhere(int position) {
        //Based on the position or someother data decide what is the next scene
        //Intent intent = new Intent(activity.get(),NextActivity.class);
        return new Intent();
    }

    @Override
    public void passDataToNextScene(int position, Intent intent) {
        //Based on the position or someother data decide the data for the next scene
        // StatementsModel flight = activity.get().listOfSomething.get(position);
        // intent.putExtra("flight",flight);
    }

}

File: app/src/main/java/com/earaujo/santander/statements/StatementsRouter.kt
package com.earaujo.santander.statements

import android.content.Intent
import com.earaujo.santander.login.LoginActivity
import java.lang.ref.WeakReference

class StatementsRouter : StatementsRouterInput {

    var activity: WeakReference<StatementsActivity>? = null

    override fun logout() {
        activity?.get()?.let {
            val intent = Intent(it, LoginActivity::class.java)
            it.startActivity(intent)
            it.finish()
        }
    }
}

interface StatementsRouterInput {
    fun logout()
}